### PR TITLE
Unify the server logs and the gokit logging.

### LINKED ISSUE #835: Difficulty Parsing Logs Consistently
While troubleshooting various services that are a part of Cortex, one item I've found is that we are unable to easily parse are the logs.  It would seem various parts of the services log in multiple ways which causes a bit of trouble as far as being able to create a reasonable logging filter that is parsable by X method of ingesting logs.

As an example, I targeted the distributor and found at least 3 ways this service put logs down (sanitized):
```
ts=2018-05-31T18:19:27.515467232Z caller=log.go:112 level=error org_id=0 msg="push error" err="rpc error: code = Code(400) desc = sample with repeated timestamp but different value for series container_fs_reads_total{container_name=\"example\", example_cluster=\"Example\", a_cluster=\"Example\", id=\"/kubepods.slice/kubepods-besteffort.slice/kubepods-besteffort-pod526740fc_3e0e_11e8_b916_02000a838eab.slice/docker-b83abfed1d197426a06778e7dcd6c0470ac60e77d5f76173793fd0025929d039.scope\", image=\"docker.io/example/example@sha256:261537e58647e604701c97765973e2c4d0627953f382883f83d2d64144803cdc\", instance=\"10.11.143.157:8443\", job=\"kubernetes-apiservers\", name=\"k8s_example_example-1-lqqtq_exampleproject_526740fc-3e0e-11e8-b916-02000a838eab_0\", namespace=\"example\", pod_name=\"example-1-lqqtq\"}; last value: 1089, incoming value: 1"
ts=2018-05-30T21:12:56.673829487Z caller=log.go:112 level=error msg="error getting path" key=collectors/ring err="Unexpected response code: 500"
time="2018-05-30T19:58:18Z" level=warning msg="Is websocket request: false\nPOST /api/prom/push HTTP/1.1\r\nHost: distributor\r\nConnection: close\r\nAccept-Encoding: gzip\r\nConnection: close\r\nContent-Encoding: snappy\r\nContent-Length: 36513\r\nContent-Type: application/x-protobuf\r\nUser-Agent: Go-http-client/1.1\r\nX-Prometheus-Remote-Write-Version: 0.1.0\r\nX-Scope-Orgid: 0\r\n\r\n"
```

In the first two examples, the various fields are inserted when needed (this is probably a noop). My main concern is comparing one of the first two examples with the last example.  It appears there's an entirely different logging mechanism all together.  

This makes it a bit difficult for various systems to parse the logs thoughtfully ensuring a consistent experience when viewing the logs for troubleshooting issues.

### PR DESCRIPTION
**Pull request by tomwilkie**
Fixes #835

Also tidies up the `prometheusLogger` (which keep metrics on number of log lines per level). Probably want to move this into common.

Depends on https://github.com/weaveworks/common/pull/96, so start there first

### CODE REVIEW

**Comment by tomwilkie** [comment]
Testing & this works nicely - we get decent logs from the servers again.

File: pkg/util/log.go [hunk 1]
```diff
@@ -1,117 +1,75 @@
...
-// Contrarily to MustNewPrometheusLogger, it returns an error to the caller in case of issue.
-// Use NewPrometheusLogger if you want more control. Use MustNewPrometheusLogger if you want a less verbose logger creation.
-func NewPrometheusLogger(al AllowedLevel) (log.Logger, error) {
+// NewPrometheusLogger creates a new instance of PrometheusLogger which exposes
+// Prometheus counters for various log levels.
+func NewPrometheusLogger(l logging.Level) (log.Logger, error) {
 	// This code copy-pasted from prometheus/common/promlog.New()
-	l := log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
-	l = al.Filter(l)
+	logger := log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
```
Comment (bboreham) [inline pkg/util/log.go]: Is the comment still correct?  ("This code copy-pasted from prometheus/common/promlog.New()")

**Comment by tomwilkie** [comment]
Rebase, pulled in new weaveworks/common so also fixes #882 (and closes #883)

File: pkg/util/log.go (hunk 1 above)
Comment (tomwilkie) [reply pkg/util/log.go]: Not in any meaningful way, no.  Will remove.

**Comment by tomwilkie** [comment]
> Also I'm not clear what replaced AllowedLevel, but that's mostly curiosity.

Thats basically replaced with https://github.com/weaveworks/common/blob/master/logging/level.go

**Comment by tomwilkie** [comment]
Can I interpret "Looks ok" as LGTM? :-)

**Comment by tomwilkie** [comment]
> I notice kuberesolver was updated; don't know what difference that makes.

Good catch.  Let me look into that, as there hasn't been semver on that repo before, and a bunch of breaking changes have gone in...

**Comment by tomwilkie** [comment]
Looks safe to me - its pre v2 (when the API changed) and pretty much only adds a change by you, Bryan!

**Comment by bboreham** [comment]
I missed that the metric `log_messages` changed to `log_messages_total`. 
The new name is more correct, but the change is disruptive.

**Pull request merged**